src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java

```diff
@@ -253,6 +260,34 @@ private ResponseEntity<DocumentDetails> getDocumentDetails(String authToken, UUI
         }
     }
 
+    /**
+     * Adds all uploaded documents to the case's document collection.
+     * @param caseData case that provides both document collections(uploaded and case doc collections)
+     */
+    public void addUploadedDocsToCaseDocCollection(CaseData caseData) {
+
+        //If doc collection is empty, initialise it
+        if (caseData.getDocumentCollection() == null) {
+            caseData.setDocumentCollection(new ArrayList<>());
+        }
+
+        caseData.getAddDocumentCollection().forEach(
+                uploadDoc -> {
+                    DocumentType uploadedDocType = uploadDoc.getValue();
+                    setDocumentTypeForDocument(uploadedDocType);
+                    setSecondLevelDocumentFromType(uploadedDocType, uploadedDocType.getDocumentType());
+                    DocumentTypeItem docTypeItem = createDocumentTypeItemFromTopLevel(
+                            uploadedDocType.getUploadedDocument(),
+                            uploadedDocType.getTopLevelDocuments(),
+                            uploadedDocType.getDocumentType(),
+                            String.format("%s : %s", uploadedDocType.getShortDescription(),
+                                    uploadedDocType.getTopLevelDocuments()));
+                    docTypeItem.getValue().setDateOfCorrespondence(uploadDoc.getValue().getDateOfCorrespondence());
+                    caseData.getDocumentCollection().add(docTypeItem);
+                    setDocumentNumbers(caseData);
```

Comment:
setDocumentNumbers method iterate through the caseData document collection. should move the method call out of the loop I think

Reply:
thanks. Done